### PR TITLE
philadelphia-core: Add 'FIXConnection#reprepare()'

### PR DESCRIPTION
**Pull request by jvirtanen**
Replace the existing `updateCompID()` method with one that feels more at home with the existing `prepare()` and `update()` methods.

### CODE REVIEW

File: examples/acceptor/src/main/java/com/paritytrading/philadelphia/acceptor/Session.java [hunk 1]
```diff
@@ -77,7 +77,7 @@ public void reject(FIXConnection connection, FIXMessage message) throws IOExcept
             public void logon(FIXConnection connection, FIXMessage message) throws IOException {
                 connection.sendLogon(true);
 
-                connection.updateCompID(report);
+                connection.reprepare(report);
```
Comment (jheusser) [inline examples/acceptor/src/main/java/com/paritytrading/philadelphia/acceptor/Session.java:80]: This definitely reads like a typo 😆

**Review by jheusser — APPROVED** [review]
Looks good to me, if the overloaded versions of `prepare` wouldn't be confusing, could also set `prepare` to public 👍🏽

File: examples/acceptor/src/main/java/com/paritytrading/philadelphia/acceptor/Session.java (hunk 1 above)
Comment (jvirtanen) [reply examples/acceptor/src/main/java/com/paritytrading/philadelphia/acceptor/Session.java:80]: I think I got the feedback I needed here. 😂

Comment (JanStureNielsen) [reply examples/acceptor/src/main/java/com/paritytrading/philadelphia/acceptor/Session.java:80]: `resetOrPrepare`? Or avoiding it with a `setOrAdd` behavior in `FIXMessage`?

**Comment by jvirtanen** [comment]
Superseded by #209.

**Pull request closed without merging**